lib/seahorn/HornClauseDBTransf.cc

```diff
@@ -58,4 +58,80 @@ void normalizeHornClauseHeads(HornClauseDB &db) {
     db.addRule(new_rule);
   }
 }
+
+template <typename Set, typename Predicate> void erase_if(Set &s, Predicate shouldRemove) {
+  for(auto it = s.begin(); it!=s.end();){
```

Comment:
this is dangerous. `erase` might invalidate the iterator. There is already `std::remove_if` that does something similar, but is safer.

You can make it safer by specializing on `Set` and commenting why it works with that particular implementation of a set.